src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/FileRenameTracker.cs

```diff
@@ -45,39 +45,44 @@ public FileRenameTracker(IUnconfiguredProjectVsServices projectVsServices, Visua
             _roslynServices = roslynServices;
         }
 
-        public async Task HintedAsync(IImmutableDictionary<Guid, IImmutableSet<IProjectChangeHint>> hints)
+        public Task HintedAsync(IImmutableDictionary<Guid, IImmutableSet<IProjectChangeHint>> hints)
         {
-            var files = hints.GetValueOrDefault(ProjectChangeFileSystemEntityRenameHint.RenamedFile) ?? ImmutableHashSet.Create<IProjectChangeHint>();
+            var files = hints[ProjectChangeFileSystemEntityRenameHint.RenamedFile];
             if (files.Count == 1)
             {
-                var hint = files.First() as IProjectChangeFileRenameHint;
-                if (hint != null && !hint.ChangeAlreadyOccurred)
+                var hint = (IProjectChangeFileRenameHint)files.First();
+                if (!hint.ChangeAlreadyOccurred)
                 {
                     var kvp = hint.RenamedFiles.First();
-                    await ScheduleRenameAsync(kvp.Key, kvp.Value).ConfigureAwait(false);
+                    ScheduleRenameAsync(kvp.Key, kvp.Value);
                 }
             }
+
+            return Task.CompletedTask;
         }
 
         public Task HintingAsync(IProjectChangeHint hint)
         {
             return Task.CompletedTask;
         }
 
-        private async Task ScheduleRenameAsync(string oldFilePath, string newFilePath)
+        private void ScheduleRenameAsync(string oldFilePath, string newFilePath)
         {
             string codeExtension = Path.GetExtension(newFilePath);
-            if (codeExtension == null || !oldFilePath.EndsWith(codeExtension, StringComparison.OrdinalIgnoreCase))
+            if (!oldFilePath.EndsWith(codeExtension, StringComparison.OrdinalIgnoreCase))
             {
                 return;
             }
 
-            await _projectVsServices.ThreadingService.SwitchToUIThread();
-
             var myProject = _visualStudioWorkspace
```

Comment:
I was under the impression that the workspace query needs to be executed on UI thread.